collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDetails.kt

```diff
@@ -22,12 +22,68 @@ data class ServerFormDetails @JvmOverloads constructor(
     val formId: String?,
     val formVersion: String?,
     val hash: String?,
-    val isNotOnDevice: Boolean,
-    val isUpdated: Boolean,
-    val manifest: ManifestFile?
+    val manifest: ManifestFile?,
+    val type: Type
 ) : Serializable {
 
+    @Deprecated(message = "Use primary constructor instead")
+    constructor(
+        formName: String?,
+        downloadUrl: String?,
+        formId: String?,
+        formVersion: String?,
+        hash: String?,
+        isNotOnDevice: Boolean,
+        isUpdated: Boolean,
+        manifest: ManifestFile?,
+    ) : this(
+        formName,
+        downloadUrl,
+        formId,
+        formVersion,
+        hash,
+        manifest,
+        if (isNotOnDevice) {
+            Type.New
+        } else if (isUpdated) {
+            Type.UpdatedVersion
+        } else {
+            Type.OnDevice
+        }
+    )
+
+    fun isUpdated(): Boolean {
+        return type != Type.OnDevice && type != Type.New
+    }
+
     companion object {
-        private const val serialVersionUID = 3L
+        private const val serialVersionUID = 4L
+    }
+
+    enum class Type {
+        /**
+         * The form is on the device already
+         */
+        OnDevice,
+
+        /**
+         * The form is not on the device
+         */
+        New,
+
+        /**
+         * The form is on the device, but this is a new version with a new version and hash
```

Comment:
`a new version with a new version` sounds weird, the second version refers to the version number, I guess?